webapp/packages/plugin-settings-panel/src/SettingsPanel/SettingsGroup.tsx

```diff
@@ -7,26 +7,33 @@
  */
 import { observer } from 'mobx-react-lite';
 
-import { Group, GroupTitle } from '@cloudbeaver/core-blocks';
+import { getComputed, Group, GroupTitle, useTranslate } from '@cloudbeaver/core-blocks';
 import type { ISettingDescription, ISettingsSource, SettingsGroup as SettingsGroupType } from '@cloudbeaver/core-settings';
+import { isArraysEqual } from '@cloudbeaver/core-utils';
+import { ITreeFilter } from '@cloudbeaver/plugin-navigation-tree';
 
 import { getSettingGroupId } from './getSettingGroupId';
 import { Setting } from './Setting';
+import { settingsFilter } from './settingsFilter';
 import { SettingsGroupTitle } from './SettingsGroupTitle';
 
 interface Props {
   group: SettingsGroupType;
   source: ISettingsSource;
-  settings: ISettingDescription<any>[];
+  settings: Map<SettingsGroupType, ISettingDescription<any>[]>;
+  treeFilter: ITreeFilter;
 }
 
-export const SettingsGroup = observer<Props>(function SettingsGroup({ group, source, settings }) {
+export const SettingsGroup = observer<Props>(function SettingsGroup({ group, source, settings, treeFilter }) {
+  const translate = useTranslate();
+  const groupSettings = getComputed(() => settings.get(group)?.filter(settingsFilter(translate, treeFilter.filter)) || [], isArraysEqual);
```

Comment:
there is no need for current specific `SettingsGroup` to know something about all other `settings` map. I suggest to calculate this in parent component (in mapping function helper maybe idk) and pass here as a prop. so we wont mess up the logic here in future

Reply:
this made for optimization, before this settings was created together with group in parent component, that lead to full re-render because of new props (always new array of settings)

current implementation shifts settings getter logic to component that renders it